File: whisper_live/enums/model_type.py

from enum import StrEnum


class ModelType(StrEnum):
    Default = "Default"
    WhisperX = "WhisperX"
    Thonburian = "Thonburian"

    @staticmethod
    def parse(text: str):
        if text == "WhisperX":
            return ModelType.WhisperX
        elif text == "Thonburian":
            return ModelType.Thonburian
        else:
            return ModelType.Default

File: dotnet/TranscriptionClient/Transcriptor.Py.Wrapper/Enums/ModelType.cs
namespace Transcriptor.Py.Wrapper.Enums;

public enum ModelType
{
    Default,
    [Obsolete("WhisperX is no longer maintained, please use `Default` with CTranslate2 model instead.")]
    WhisperX,
    [Obsolete("Thonburian is no longer maintained, please use `Default` with CTranslate2 model instead.")]
    Thonburian,
}